src/main/java/com/devonfw/quarkus/general/service/exception/RestServiceExceptionFacade.java

```diff
@@ -0,0 +1,93 @@
+package com.devonfw.quarkus.general.service.exception;
+
+import java.time.ZonedDateTime;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Provider
+@Slf4j
+public class RestServiceExceptionFacade implements ExceptionMapper<RuntimeException> {
+
+  private boolean exposeInternalErrorDetails = false;
+
+  @Context
+  UriInfo uriInfo;
+
+  @Override
+  public Response toResponse(RuntimeException exception) {
+
+    log.error("Exception:{},URL:{},ERROR:{}", exception.getClass().getCanonicalName(), this.uriInfo.getRequestUri(),
+        exception.getMessage());
+
+    if (exception instanceof ApplicationBusinessException) {
+      return createResponse((ApplicationBusinessException) exception);
+    }
+    if (exception instanceof WebApplicationException) {
```

Comment:
The Quarkus RESTEasy extension provides a mapper for `NotFoundException` out of the box, that produces a 404 response (https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java).
Should we anyway implement and use an own mapper for it?

Reply:
If that already plays together with out exception mapper, it is fully sufficient. But then you could add a nice inline comment to the code giving this valuable information including that link so people can easily understand what is going on. I also was not aware of that so thanks for making this clear.

Reply:
But comming back to my original concern:
How about permission denied? How about validation failures (from BV)?
We do not want to report this as 500 (internal server error)!
This is IMHO only complete if we have all these cases covered.
Ideally we even add a JUnit that is testing all these scenarios to make them transparent and to document our features and behaviour.

Reply:
Since the Quarkus ExceptionMapper implementations do not return any further information (only status code) I decided to create own implementations. So now we have exception mapper for NotFoundException, UnauthorizedException and ValidationException + our ApplicationBusinessException. Other exceptions are handled as WebApplicationException or as general Exception with status 500
@hohwille is this what you had in mind?